test/Distributed/Runtime/distributed_actor_remote_functions.swift

```diff
@@ -155,21 +153,21 @@ func test_remote_invoke(address: ActorAddress, transport: ActorTransport) async
     if __isRemoteActor(actor) {
       do {
         _ = try await actor.helloThrowsTransportBoom()
-        preconditionFailure("helloThrowsTransportBoom: should have thrown")
+        print("WRONG: helloThrowsTransportBoom: should have thrown")
       } catch {
         print("\(personality) - helloThrowsTransportBoom: \(error)")
       }
     } else {
       do {
         _ = try await actor.helloThrowsImplBoom()
-        preconditionFailure("helloThrowsImplBoom: Should have thrown")
+        print("WRONG: helloThrowsImplBoom: Should have thrown")
```

Comment:
run the test and notice that our current impl with the `      if (!selfApply) {` simply causes us to NEVER invoke the  thunk, which is super bad.